frontend/javascripts/oxalis/view/settings/dataset_settings_view.js

```diff
@@ -76,9 +78,76 @@ type DatasetSettingsProps = {|
   onChangeUser: (key: $Keys<UserConfiguration>, value: any) => void,
   onUnlinkFallbackLayer: Tracing => Promise<void>,
   tracing: Tracing,
+  task: ?Task,
 |};
 
-class DatasetSettings extends React.PureComponent<DatasetSettingsProps> {
+function DownsampleVolumeModal({ visible, hideDownsampleVolumeModal, magsToDownsample }) {
+  const [isDownsampling, setIsDownsampling] = useState(false);
+
+  const handleTriggerDownsampling = async () => {
+    setIsDownsampling(true);
+    await api.tracing.downsampleSegmentation();
+    setIsDownsampling(false);
+  };
+
+  return (
+    <Modal
+      title="Downsample Volume Annotation"
+      onCancel={isDownsampling ? null : hideDownsampleVolumeModal}
+      visible={visible}
+      footer={null}
+      width={800}
+      maskClosable={false}
+    >
+      <p>
+        This annotation does not have volume annotation data in all resolutions. Consequently,
+        annotation data cannot be rendered at all zoom values. By clicking &quot;Downsample&quot;,
+        webKnossos will use the best resolution of the volume data to create all dependent
+        resolutions.
+      </p>
+
+      <p>
+        The following resolutions will be added when clicking &quot;Downsample&quot;:{" "}
+        {magsToDownsample.map(mag => mag.join("-")).join(", ")}.
+      </p>
+
+      <p>
+        The cause for the missing resolutions can be one of the following:
+        <ul>
+          <li>
+            The annotation was created before webKnossos supported multi-resolution volume tracings.
+          </li>
+          <li>An old annotation was uploaded which did not include all resolutions.</li>
+          <li>The annotation was created in a task that was restricted to certain resolutions.</li>
+          <li>The dataset was mutated to have more resolutions.</li>
+        </ul>
+      </p>
+
+      <p style={{ fontWeight: "bold" }}>
+        Note that this action might take a few minutes. Afterwards, the annotation is reloaded.
+        Also, the version history of the volume data will be reset.
```

Comment:
Out of curiosity, why does the history need to be reset?

Reply:
I think, the entire `tracing` object within the annotation is swapped out with a new one. Since history is only persisted for a tracing, such swap-outs cannot be recorded. The reason for swapping out the entire volume tracing is probably, since the new tracing needs to have a different resolution set?

Correct me if I'm wrong, @fm3!

Obviously, it would be cool if we could keep the history. @fm3 Would this be feasible for a later iteration?

Reply:
Yes, that’s correct, the downsample call is a variant of duplicate. This was the easiest way to implement this and ensure no version conflicts etc. However, it does not have to be done like this, a future iteration could probably implement the downsampling as writing new version in an existing volume tracing.

The backend stdout logging also includes both the old and the new tracing id along with the annotation id if this happens, so if a user *really* needs to undo this, we can do it for them in the db.